src/modules/types.rs

```diff
@@ -39,10 +58,19 @@ pub fn parse_type(meta: &mut ParserMetadata) -> Result<Type, Failure> {
 
 // Tries to parse the type - if it fails, it fails quietly
 pub fn try_parse_type(meta: &mut ParserMetadata) -> Result<Type, Failure> {
+    let mut is_failable = false;
     let tok = meta.get_current_token();
-    match tok.clone() {
-        Some(matched_token) => {
-            match matched_token.word.as_ref() {
+    let word = tok.clone().map(|tok| tok.word);
+    let word = word.as_deref().map(|word| match word.strip_suffix('?') {
+        Some(name) => {
+            is_failable = true;
+            name
+        },
+        _ => word,
+    });
```

Comment:
I think that the `?` symbol should be parsed as a separate token. I'm surprised that this wasn't a thing. The fact that in case of command and function call it was separated was because these two are symbols and are too separated. Could you add this symbol to the [Heraclitus rules](https://github.com/amber-lang/amber/blob/571f77ca9955832693108df60e36d79145df18d9/src/rules.rs#L4)?

This will require to modify the selected code a bit. Just check for the next token if it's `tok.word == "?"` and that's it.